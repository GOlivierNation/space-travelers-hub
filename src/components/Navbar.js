import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../assets/image/logo.png';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Rockets',
    },
    {
      id: 2,
      path: '/missions',
      text: 'Missions',
    },
    {
      id: 3,
      path: '/myprofile',
      text: 'MyProfile',
    },
  ];
  return (
    <nav>
      <img src={logo} alt="logo" />
      <ul>
        {links.map((link) => (
          <li key={link.id}>
            <NavLink
              to={link.path}
              className={({ isActive }) => (isActive ? 'active' : '')}
            >
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};
export default Navbar;
