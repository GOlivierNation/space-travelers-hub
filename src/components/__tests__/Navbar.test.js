import React from 'react';
import '@testing-library/jest-dom';
import { fireEvent, render, screen,  } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Navbar';

describe('Tests for the NavBar', () => {
  test('Should render the Missions', () => {
    render(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(screen.getByText(/Missions/)).toBeInTheDocument();
  });

  test('Should render the Rockets', () => {
    render(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(screen.getByText(/Rockets/)).toBeInTheDocument();
  });

  test('Should render the MyProfile', () => {
    render(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(screen.getByText(/MyProfile/)).toBeInTheDocument();
  });

  test('When click on Missions sholud change the link to that page', () => {
    render(<BrowserRouter><Navbar /></BrowserRouter>);
    fireEvent.click(screen.getByText(/Missions/));
    expect(window.location.href).toBe(screen.getByText(/Missions/).href);
  });

  it('Should render the Navbar', () => {
    const tree = renderer.create(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(tree).toMatchSnapshot();
  });
});