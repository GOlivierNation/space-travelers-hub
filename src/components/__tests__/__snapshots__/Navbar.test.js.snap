// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the NavBar Should render the Navbar 1`] = `
<nav>
  <div
    className="nav-header"
  >
    <img
      alt="logo"
      src="logo.png"
    />
    <h3>
      Space Travelers' Hub
    </h3>
  </div>
  <ul>
    <li>
      <a
        className=""
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li>
      <a
        className=""
        href="/myprofile"
        onClick={[Function]}
      >
        MyProfile
      </a>
    </li>
  </ul>
</nav>
`;
